pkg/configuration/configuration.go

```diff
@@ -84,6 +90,7 @@ type extendedViper struct {
 	mutex               sync.RWMutex
 	automaticEnvEnabled bool
 	configFiles         []string
+	defaultCache        *cache.Cache
```

Comment:
question: wouldn't it be better to define our own cache interface here? This seems quite tightly coupled with the caching package we're using.